refactor/response-list

ahmedNaser7 opened the pull request:
Refactor: Remove Response entity from lists module

 removes the `Response` data class from the `domain.lists.entity` package and updates all related use cases, repositories, and tests to use a simple `Boolean` return type instead.

The `Response` class previously encapsulated a status code, status message, and success boolean. This change simplifies the return type of list manipulation operations (create, delete, add movie, remove movie) to directly reflect success or failure with a boolean.

Specifically:
- Deleted `domain/lists/src/main/java/com/paris/domain/lists/entity/Response.kt`.
- Updated `CreateListUseCase`, `DeleteListUseCase`, `AddMovieToListUseCase`, and `RemoveMovieFromListUseCase` to return `Boolean`.
- Updated `ListsRepository` interface and `ListsRepositoryImpl` implementation to reflect the change in return type for list manipulation methods.
- Modified tests for the aforementioned use cases and repository implementation to work with the new `Boolean` return type.
- Adjusted `ListsViewModel` to use the boolean result directly for snackbar success status.
- Removed the `toDomain()` mapping for `ResponseDto` in `ListsMapper.kt`.

ahmedNaser7 committed:
Refactor: Remove Response entity from lists module

This commit removes the `Response` data class from the `domain.lists.entity` package and updates all related use cases, repositories, and tests to use a simple `Boolean` return type instead.

The `Response` class previously encapsulated a status code, status message, and success boolean. This change simplifies the return type of list manipulation operations (create, delete, add movie, remove movie) to directly reflect success or failure with a boolean.

Specifically:
- Deleted `domain/lists/src/main/java/com/paris/domain/lists/entity/Response.kt`.
- Updated `CreateListUseCase`, `DeleteListUseCase`, `AddMovieToListUseCase`, and `RemoveMovieFromListUseCase` to return `Boolean`.
- Updated `ListsRepository` interface and `ListsRepositoryImpl` implementation to reflect the change in return type for list manipulation methods.
- Modified tests for the aforementioned use cases and repository implementation to work with the new `Boolean` return type.
- Adjusted `ListsViewModel` to use the boolean result directly for snackbar success status.
- Removed the `toDomain()` mapping for `ResponseDto` in `ListsMapper.kt`.

ahmedNaser7 committed:
Delete MainActivity.kt from onboardingUi module

The `MainActivity.kt` file within the `feature/onboarding/onboardingUi` module has been removed.

ahmedNaser7 committed:
Merge branch 'develop' into refactor/response-list

ahmedNaser7 committed:
Refactor: Use Boolean directly instead of Response data class

The `Response` data class, which contained a boolean `success` field along with status code and message, has been removed.

Code that previously used `Response.success` now directly uses a `Boolean` value to indicate success or failure. This simplifies the codebase by removing an unnecessary data class.

ahmedNaser7 committed:
Refactor RLE and update app version to 1.0.22

Joseph-Sameh-0 commented on repository/lists/src/main/java/com/repository/lists/mapper/ListsMapper.kt:
```diff
@@ -16,10 +14,10 @@ fun ListsDto.toDomain(): List<Lists> {
-    )
-}
\ No newline at end of file
+//fun ResponseDto.toDomain(): Response {
+//    return Response(
+//        statusCode = this.statusCode?: 0,
+//        statusMessage = this.statusMessage.orEmpty(),
+//        success = this.success  ?: false
+//    )
+//}
```
don't upload commented code

ahmedNaser7 committed:
Update ListsMapper.kt

ahmedNaser7 committed:
Apply no-op drawWithContent to Shimmerable

The `drawWithContent{}` modifier is added to the `shimmerable` function in `Shimmerable.kt`. This appears to be a no-operation change, as it simply draws the existing content without modification. This might be a workaround for a recomposition issue or a placeholder for future drawing logic.

ahmedNaser7 committed:
Merge remote-tracking branch 'origin/refactor/response-list' into refactor/response-list
